### PR TITLE
Jan2 move presets from layout helper

### PR DESCRIPTION
**Pull request by IzaacAyelin**
Refactored the function that set styles by layouts (getStyleByLayout in layoutHelper) and moved the fixed styles to the correct preset in src/components/gallery/presets folder.
Added missing preset ('empty' layout) in the presets folder.

### CODE REVIEW

File: packages/gallery/src/components/gallery/presets/columnGallery.js [hunk 1]
```diff
@@ -2,21 +2,44 @@
...
+  isSlideshow: false,
+  cropOnlyFill: false,
 }
 export default class ColumnGallery extends React.Component {
 
-    render() {
-        return (
-            <ProGallery
-                {...this.props}
-                styles={{
```
Comment (IzaacAyelin) [inline packages/gallery/src/components/gallery/presets/columnGallery.js:15]: use function to build styles

File: packages/gallery/src/components/helpers/layoutHelper.js [hunk 2]
```diff
@@ -521,7 +217,18 @@ function addLayoutStyles(styles) {
...
+    const layoutName = getLayoutName(Number(styles.galleryLayout));
+    if (utils.isUndefined(layoutName) || styles.galleryLayout === '') {
+      styles = Object.assign({}, emptyLayout, styles,
+        {
+          ...NEW_PRESETS.collage,
+          gallerySize: Math.round(30 * 5 + 500)
+        });
+    }
+
+    styles = Object.assign({}, emptyLayout, styles); //legacy layouts
```
Comment (IzaacAyelin) [inline packages/gallery/src/components/helpers/layoutHelper.js]: delete comment

File: packages/gallery/src/components/gallery/presets/alternateGallery.js [hunk 3]
```diff
@@ -5,21 +5,47 @@ import LAYOUTS from '../../../common/constants/layout';
...
+  smartCrop: false,
+  collageDensity: 0.48,
+  galleryMargin: 0,
+  floatingImages: 0,
+  cubeRatio: 1,
+  fixedColumns: 1,
+  groupsPerStrip: 0,
+  oneRow: false,
+  placeGroupsLtr: false,
+  at: 1538490828979,
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/alternateGallery.js]: remove the `at`, `sampleSize`

File: packages/gallery/src/components/gallery/presets/emptyGallery.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+
+import React from 'react';
+import ProGallery from '../proGallery/proGallery';
+import LAYOUTS from '../../../common/constants/layout';
+
+export const fixedStyles = {
+  galleryLayout: LAYOUTS.EMPTY,
+}
+export default class ColumnGallery extends React.Component {
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/emptyGallery.js]: EmptyGallery

File: packages/gallery/src/components/gallery/presets/gridGallery.js [hunk 5]
```diff
@@ -4,18 +4,46 @@ import LAYOUTS from '../../../common/constants/layout';
...
+  
+  //this params were moved from the presets in layoutHelper and were not tested and checked yet.
+  showArrows: false,
+  cubeImages: true,
+  smartCrop: false,
+  isVertical: true,
+  galleryType: 'Columns',
+  groupSize: 1,
+  groupTypes: '1',
+  fixedColumns: undefined,
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/gridGallery.js]: use 0 instead

File: packages/gallery/src/components/gallery/presets/masonryGallery.js [hunk 6]
```diff
@@ -5,19 +5,43 @@ import LAYOUTS from '../../../common/constants/layout';
...
-                    ...this.props.styles,
-                    ...fixedStyles
-                }}
-            />
-        );
+  createStyles = () => {
+    return {
+      ...this.props.styles,
+      ...fixedStyles,
+      gallerySize: this.props.styles.gallerySize,
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/masonryGallery.js:32]: It was like this

File: packages/gallery/src/components/gallery/presets/panoramaGallery.js [hunk 7]
```diff
@@ -2,20 +2,48 @@
...
+      ...this.props.styles,
+      ...fixedStyles,
     }
+  }
+
+  render() {
+    return (
+      <ProGallery
+        {...this.props}
+        styles={{
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/panoramaGallery.js]: do not double spread

File: packages/gallery/src/components/gallery/presets/thumbnailGallery.js [hunk 8]
```diff
@@ -2,21 +2,52 @@
...
-  enableInfiniteScroll: true
+  enableInfiniteScroll: true,
+  
+  //this params were moved from the presets in layoutHelper and were not tested and checked yet.
+  showArrows: true,
+  cubeImages: true,
+  smartCrop: false,
+  cubeRatio: '100%/100%',
+  isVertical: false,
+  galleryType: 'Strips',
```
Comment (guysopher) [inline packages/gallery/src/components/gallery/presets/thumbnailGallery.js:17]: this param can be removed

File: packages/gallery/src/components/helpers/layoutHelper.js [hunk 9]
```diff
@@ -44,88 +44,6 @@ const emptyLayout = {
...
-  }
-  style.galleryType = style.isVertical ? 'Columns' : 'Strips';
-  style.groupSize = parseInt(style.groupTypes.slice(-1)[0]);
-  style.groupTypes = style.groupTypes.join(',');
-  style.minItemSize = style.gallerySize / style.groupSize / 2;
-
-  return style;
-}
 
 function getStyleByGalleryType(styles) {
```
Comment (guysopher) [inline packages/gallery/src/components/helpers/layoutHelper.js:47]: ask Emanuelle to add this styleParam (galleryType) to the dashboard

File: packages/gallery/src/components/helpers/layoutHelper.js [hunk 10]
```diff
@@ -473,26 +188,7 @@ const getLayoutName = (galleryLayout) => {
...
-
-  if (utils.isVerbose()) {
-    console.log('chosen layout is', layoutName);
-  }
-  const layouts = getPreset(styles)
-  return {
-    ...layouts[layoutName](),
-    galleryLayout
-  };
+  return galleyLayoutList[galleryLayout + 1]
```
Comment (guysopher) [inline packages/gallery/src/components/helpers/layoutHelper.js:190]: use the gallery consts

File: packages/gallery/src/components/helpers/layoutHelper.js [hunk 11]
```diff
@@ -521,7 +217,18 @@ function addLayoutStyles(styles) {
     if (utils.isVerbose()) {
       console.log('Using galleryLayout for defaults', styles);
     }
-    styles = Object.assign({}, emptyLayout, styles, getStyleByLayout(styles)); //legacy layouts
+
```
Comment (guysopher) [inline packages/gallery/src/components/helpers/layoutHelper.js:219]: this code should be irrelevant after choosing preset

**Pull request closed without merging**